docs/general/ddl/data-types.rst

```diff
@@ -347,6 +351,66 @@ In these expressions, the desired time zone is specified as a string
 The scalar function :ref:`TIMEZONE <scalar-timezone>` (zone, timestamp) is
 equivalent to the SQL-conforming construct timestamp AT TIME ZONE zone.
 
+.. _time-data-type:
+
+time with time zone
+-------------------
+
+The time type consist of time followed by an optional time zone.
+
+``timetz`` is a valid alias for `time with time zone`.
+
+Internally, time values are mapped to microseconds since midnight and seconds
+as a signed offset from UTC for the time zone, that is a `bigint` followed by
+an `integer`, or 12 bytes.
```

Comment:
```suggestion
```

The storage / range is already present in the table further above, we don't need to document further internals. (Would also restrict us in making changes in the future ..)